### DISCUSSION TITLE
Linux: Wayland support ?

### DISCUSSION THREAD

**Discussion by renueman** [Ylianst/MeshCentral#6316] [Issues]
I am previewing Linux Mint 22 in a VM and it is running wayland. I have installed the meshagent but it does not show up in devices. The service is running and I do not see anything that sticks out to me except wayland is running instead of Xorg. Im wondering if that might be the issue.
Is there a log file for meshagent in linux? 
I am running 1.1.0 stable and I see 1.1.27 is listed as the recent. Is it safe to upgrade?

**Comment by marclaporte** [comment]
We need help for Wayland support:
- https://github.com/Ylianst/MeshAgent/issues/115
- https://github.com/Ylianst/MeshCentral/issues/2719

**Reply by renueman** [reply]
I spent some time researching this and this is what I have found: Using a VM of LM22 - fresh install with updates. 
1. Installed using the 'add agent' in a group. I copied the command line to clipboard and ran as root.  Checking status of the meshagent service showed it running but refused to show up in the device group. 
2. I uninstalled meshagent and then re-installed via the binary. This time it showed up. I connected to it and it ran fine. I could remote into the VM of LM22. 

I ran a system report to check the graphics screen settings and I see 'xwayland' 

> Display: x11 server: X.Org v: 21.1.11 with: Xwayland v: 23.2.6 compositor: marco v: 1.26.2
>     driver: X: loaded: vmware unloaded: fbdev,modesetting,vesa dri: swrast gpu: vmwgfx display-ID: :0
>     screens: 1

Some reading on xwayland discribes it as a backwards compatible service for Xorg. So, That solves this issue. I have no idea why the meshcentral command line does not show up in the device group but the install of the binary instead does show up in the device group.

Thank you for your replies and links. I hope wayland support happens soon but at least xwayland is a working solution right now. :)

**Reply by ne0YT** [reply]
@renueman I use https://system76.com/pop/ "PoP! OS" and it does not work even though i have xwayland installed.
is there something else to do to make the desktop work instead of getting the error?:
```
This platform is configured to use Xwayland
please modify config to use Xorg
```
